lib/compliance_data.rb

```diff
@@ -0,0 +1,56 @@
+require 'aws-sdk'
+
+class ComplianceData
+  attr_reader :bucket
+
+  def initialize(settings)
+    raise ArgumentError if settings.nil?
+    Aws.config.update(
+      region: settings.aws_region,
+      credentials: Aws::Credentials.new(settings.aws_access_key,
+                                        settings.aws_secret_key))
+    @settings = settings
+    @bucket = Aws::S3::Bucket.new(settings.aws_bucket)
+  end
+
+  def base_name(full_name)
+    File.basename (full_name || ''), @settings.results_format
+  end
+
+  def full_name(base_name)
+    return '' if base_name.nil?
+    "#{@settings.results_folder}/#{base_name}#{@settings.results_format}"
+  end
+
+  def key_list
+    projects = []
+    (@bucket.objects(prefix: @settings.results_folder) || []).each do |project|
+      projects.push base_name(project.key)
+    end
+    projects
+  end
+
+  def get_version_list(name)
```

Comment:
In Ruby, it's convention for the `get` part of a method name to be implicit, and use a pluralized word to indicate that it returns an `Enumerable`. In other words, `#get_version_list` would be `#versions`, `#get_file` would be `#file` or `#file_for`, etc. @ctro Do you agree?